### PR TITLE
Added model validations

### PR DESCRIPTION
**Pull request by DevKaranPradhan**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+  validates :email, uniqueness: true, format: {with: /\A([\w+\-].?)+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i, message: "invalid format"}
+  
   has_and_belongs_to_many :matches
   has_many :user_city_sports
+  validates_associated :matches, :user_city_sports
+
+  before_save do
+    self.first_name = self.first_name.titleize
+    self.last_name = self.last_name.titleize
+  end
```
Comment (tanya-saroha) [inline app/models/user.rb]: @DevKaranPradhan Can you please explain what you are trying to do here?

File: app/models/city.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
 class City < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+
   has_many :city_sports
+  validates_associated :city_sports
```
Comment (tanya-saroha) [inline app/models/city.rb]: @DevKaranPradhan For what purpose are you validating associated records here?

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
+  validates :email, uniqueness: true, format: {with: /\A([\w+\-].?)+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i, message: "invalid format"}
+  
   has_and_belongs_to_many :matches
   has_many :user_city_sports
+  validates_associated :matches, :user_city_sports
+
+  before_save :titleize
+  
+  def titleize
+    self.first_name = self.first_name.titleize
```
Comment (tanya-saroha) [inline app/models/user.rb]: @DevKaranPradhan  Titleize what? Shouldn't it be "titleize_names"?

Comment (DevKaranPradhan) [reply app/models/user.rb]: done

**Pull request merged**